indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -203,8 +284,39 @@ class RHRegistrationEdit(RHManageRegistrationBase):
     """Edit the submitted information of a registration"""
 
     def _process(self):
+        form = make_registration_form(self.regform, management=True, edit=True)()
+
+        if form.validate_on_submit():
+            modify_registration(self.registration, form.data, self.event)
+            return redirect(url_for('.registration_details', self.registration))
+        elif form.is_submitted():
+            # not very pretty but usually this never happens thanks to client-side validation
+            for error in form.error_list:
+                flash(error, 'error')
+
+        section_data = get_event_section_data(self.regform, management=True)
+        registration_data = {r.field_data.field.html_field_name: r.data for r in self.registration.data}
+        data_by_html_name = {r.field_data.field.html_field_name: r.field_data for r in self.registration.data}
+
+        for section in section_data:
+            for field in section['items']:
+                if 'choices' in field:
```

Comment:
As I said, don't be too harsh on this commit. I'm planning on moving this at least to a function or something.